compiler/rustc_mir_transform/src/validate.rs

```diff
@@ -898,8 +898,8 @@ impl<'a, 'tcx> Visitor<'tcx> for TypeChecker<'a, 'tcx> {
                         self.param_env,
                         adt_def.non_enum_variant().fields[field].ty(self.tcx, args),
                     );
-                    if fields.len() == 1 {
-                        let src_ty = fields.raw[0].ty(self.body, self.tcx);
+                    if let [field] = &*fields.raw {
```

Comment:
:( this is kinda annoying to have to do, but I guess at least equally annoying as `fields.raw[0]`.